### PR TITLE
Added attachment style options

### LINKED ISSUE #2232: Card backgrounds should be different than text backgrounds
## Feature Request

I am noticing that when I change the text background the card background changes the same.  These should 2 entities should be separate in their core styling.  This is very important.  If I change background color, border and radius that should not also affect my adaptive card background.  

would you be able to seperate the concerns?

[Enhancement]

### PR DESCRIPTION
**Pull request by stevkan**
Fixes #2232 

## Changelog Entry

Fix [#2232](https://github.com/microsoft/BotFramework-WebChat/issues/2232). Added attachment style options, by [@stevkan](https://github.com/stevkan) in PR [#2410](https://github.com/microsoft/BotFramework-WebChat/pull/2410)

## Description

Provides the ability to apply separate styling options for attachments (cards) and activities sent from the bot. Addresses attachment background and text color, attachment border styling, and nub border styling.

## Specific Changes

Updated defaultStyleOptions.js:
- Included style options for attachments.
- Altered 'bot' property names to include "FromBot". For example, changed `bubbleBackground` to `bubbleFromBotBackground`.

Updated adaptiveCardHostConfig.js:
- Included `backgroundColor` in `containerStyles` styles.
- Set some `containerStyles` values to `null`. Otherwise, values compile as inline CSS requiring `!important` to be appended to style options in order to override.
  - As default values are providedd via the defaultStyleOptions.js, including `null` values should not have an effect.

Propagated property name changes across project files.

---

-  [ ] Testing Added
Unable to test due to PowerShell/Docker issues. Will require another to perform testing.

### CODE REVIEW

File: __tests__/bubbleNub.js [hunk 1]
```diff
@@ -43,16 +45,18 @@ describe('bubble nub', () => {
   beforeEach(() => {
     props = {
       styleOptions: {
-        bubbleBorderColor: 'red',
```
Comment (tdurnford) [inline __tests__/bubbleNub.js]: We can't remove`bubbleBorder*` from the style options because customers might be using these options. Removing them would be a breaking change. Instead set the default value to `undefined`  and in the create style set function use `bubbleBorder*` if it's provided; otherwise, use `bubbleFromBotBorder*`

File: __tests__/bubbleNub.js [hunk 2]
```diff
@@ -43,16 +45,18 @@ describe('bubble nub', () => {
...
+        bubbleFromBotBorderRadius: 10,
+        bubbleFromBotBorderWidth: 2,
         bubbleFromUserBorderColor: 'green',
         bubbleFromUserBorderRadius: 10,
         bubbleFromUserNubOffset: 0,
         bubbleFromUserNubSize: 10,
+        bubbleFromUserNubBorderColor: 'white',
         bubbleFromUserBorderWidth: 2,
-        bubbleNubOffset: 0,
-        bubbleNubSize: 10
```
Comment (tdurnford) [inline __tests__/bubbleNub.js]: Same thing as above

File: packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
+    '& p': {
+      color: bubbleAttachmentTextColor
+    },
+
     '& .ac-pushButton': {
       backgroundColor: 'White',
       borderStyle: 'solid',
       borderWidth: 1,
-      color: accent,
+      color: 'accent',
```
Comment (tdurnford) [inline packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js]: Accent shouldn't be in quotes.

File: packages/bundle/src/adaptiveCards/Styles/adaptiveCardHostConfig.js [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
         },
         accent: {
-          default: accent,
+          default: null,
           subtle: accent
         }
       }
     },
     emphasis: {
-      backgroundColor: cardEmphasisBackgroundColor,
```
Comment (tdurnford) [inline packages/bundle/src/adaptiveCards/Styles/adaptiveCardHostConfig.js]: Why is this being removed. I think there are a lot of unnecessary changes to host config.

File: packages/component/src/Styles/defaultStyleOptions.js [hunk 5]
```diff
@@ -5,14 +5,13 @@ function fontFamily(fonts) {
 }
 
 const DEFAULT_ACCENT = '#0063B1';
-const DEFAULT_SUBTLE = '#767676'; // With contrast 4.5:1 to white
 const PADDING_REGULAR = 10;
+const DEFAULT_SUBTLE = '#767676'; // With contrast 4.5:1 to white
```
Comment (tdurnford) [inline packages/component/src/Styles/defaultStyleOptions.js]: Why did this like get moved? Should be in alphabetical order

File: packages/component/src/Styles/defaultStyleOptions.js [hunk 6]
```diff
@@ -33,26 +32,34 @@ const DEFAULT_OPTIONS = {
   userAvatarInitials: '',
 
   // Bubble
-  bubbleBackground: 'White',
```
Comment (tdurnford) [inline packages/component/src/Styles/defaultStyleOptions.js]: Removing style options is a breaking change. We want to support the legacy options as well as the new ones. Instead of removing, `bubble*`, the values should be set to `undefined`, and then in the style options we should use `bubble*` if it's not falsy; otherwise, use `bubbleFromBot*`.

File: __tests__/bubbleNub.js [hunk 7]
```diff
@@ -28,10 +28,12 @@ describe('bubble nub', () => {
     const { driver, pageObjects } = await setupWebDriver({
       props: {
         styleOptions: {
-          bubbleNubOffset: 0,
```
Comment (tdurnford) [inline __tests__/bubbleNub.js]: You should be adding new tests instead of modifying the old ones.

File: packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js (hunk 3 above)
Comment (stevkan) [reply packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js]: Was testing a feature, forgot to change back. Will correct.

File: packages/bundle/src/adaptiveCards/Styles/adaptiveCardHostConfig.js (hunk 4 above)
Comment (stevkan) [reply packages/bundle/src/adaptiveCards/Styles/adaptiveCardHostConfig.js]: I believe you are referring to cardEmphasisBackgroundColor(?). As I mention in the PR description, setting actual values (vs null) in the HostConfig compiles to inline CSS on those elements. This then requires the developer to have to append '!important' to any styling changes for them to apply as inline has higher priority.
As these values are set in the defaultStyleOptions file, it's actually redundant and unnecessary to specify them here, as well. Plus (as mentioned), it amounts to extra work for the dev.

File: packages/component/src/Styles/defaultStyleOptions.js (hunk 5 above)
Comment (stevkan) [reply packages/component/src/Styles/defaultStyleOptions.js]: I'll move it back. Didn't look close enough. Thought it said 'DEFAULT_REGULAR'.

File: __tests__/bubbleNub.js (hunk 7 above)
Comment (stevkan) [reply __tests__/bubbleNub.js]: Noted!

**Comment by corinagum** [comment]
Ping on this PR, how's it going @stevkan? 

(@CoHealer for FYI)

**Comment by corinagum** [comment]
Closing due to inactivity. Feel free to resume discussion on #2232
